spring-cloud-dataflow-module-deployers/spring-cloud-dataflow-module-deployer-kubernetes/src/main/java/org/springframework/cloud/dataflow/module/deployer/kubernetes/KubernetesModuleDeployer.java

```diff
@@ -0,0 +1,306 @@
+
+		try {
+			// The only really interesting status is coming from the containers and pods.
+			// The service and the RC don't have "realtime" status info.
+			PodList list = kubeClient.pods().withLabels(createIdMap(id)).list();
+			return buildModuleStatus(id, list);
+		}
+		catch (KubernetesClientException e) {
+			logger.warn(e.getMessage(), e);
+			return buildModuleStatus(id, null);
+		}
+	}
+	
+	@Override
+	public Map<ModuleDeploymentId, ModuleStatus> status() {
+		Map<ModuleDeploymentId, ModuleStatus> result = new HashMap<>();
+		
+		ReplicationControllerList list = 
+				kubeClient.replicationControllers()
+				.withLabel(SPRING_MARKER_KEY, SPRING_MARKER_VALUE).list();
+		
+		for (ReplicationController rc : list.getItems()) {
+			Map<String, String> labels = rc.getMetadata().getLabels();			
+			String group = labels.get(SCSM_GROUP_KEY);
+			String label = labels.get(SCSM_LABEL_KEY);			
+			ModuleDeploymentId id = new ModuleDeploymentId(group, label);
+			
+			PodList pods = kubeClient.pods().withLabels(labels).list();			
+			result.put(id, buildModuleStatus(id, pods));	
+		}		
+		return result;		
+	}
+
+	private ReplicationController createReplicationController(ModuleDeploymentId id, ModuleDeploymentRequest request, Integer externalPort) {
+        ReplicationController rc = new ReplicationControllerBuilder()
+                .withNewMetadata()
+                	.withName(createKubernetesName(id))  // does not allow . in the name
+                	.withLabels(createIdMap(id))
+                	.addToLabels(SPRING_MARKER_KEY, SPRING_MARKER_VALUE)
+            		.addToLabels(SCSM_ARTIFACT_ID, request.getCoordinates().getArtifactId())
+            		.addToLabels(SCSM_GROUP_ID, request.getCoordinates().getGroupId())
+            		.addToLabels(SCSM_VERSION, request.getCoordinates().getVersion())
+            		.addToLabels(SCSM_EXTENSION, request.getCoordinates().getExtension())
+                .endMetadata()
+                .withNewSpec()
+                	.withReplicas(request.getCount())
+                	.withSelector(createIdMap(id))
+                .withNewTemplate()
+                	.withNewMetadata()
+                		.withLabels(createIdMap(id))
+                    	.addToLabels(SPRING_MARKER_KEY, SPRING_MARKER_VALUE)
+                	.endMetadata()
+	                .withSpec(createPodSpec(request, externalPort))	                	
+                .endTemplate()
+                .endSpec().build();        
+       
+        return kubeClient.replicationControllers().create(rc);        
+	}
+
+	private PodSpec createPodSpec(ModuleDeploymentRequest request, Integer port) {
+		ContainerBuilder container = new ContainerBuilder();
+	
+		// Add default image or override with customer image
+		if (properties.getModuleLauncherImage() == null) {
+			container.withImage(DEFAULT_IMAGE_NAME);			
+		} else {
+			container.withImage(properties.getModuleLauncherImage());
+		}		
+		container.withName(CONTAINER_NAME)
```

Comment:
Where does that name appear? Could it be dynamic and derived from the deployment request?